presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -141,6 +143,15 @@ public static Optional<Long> resolveSnapshotIdByName(Table table, IcebergTableNa
         return tryGetCurrentSnapshot(table).map(Snapshot::snapshotId);
     }
 
+    public static long getSnapshotIdAsOfTime(Table table, long millisUtc)
+    {
+        return table.history().stream()
+                .filter(logEntry -> logEntry.timestampMillis() <= millisUtc)
+                .max(comparing(HistoryEntry::timestampMillis))
+                .orElseThrow(() -> new PrestoException(ICEBERG_INVALID_TABLE_TIMESTAMP, format("No history found based on timestamp for table %s", table.name())))
+                .snapshotId();
```

Comment:
When we query from an empty table using ``for timestamp as of CURRENT_TIMESTAMP``, should we throw an exception, or get an empty result?

Reply:
I think its database specific. I can change to empty result later based on customer issue or if we want to match with specific database. Error message looks meaningful since user has requested version/history.

Reply:
Well, seem trino has the same behavior.